server/src/test/java/org/opensearch/index/engine/IngestionEngineTests.java

```diff
@@ -128,6 +130,22 @@ public void testRecovery() throws IOException {
         waitForResults(ingestionEngine, 4);
     }
 
+    public void testPushAPIFailures() throws IOException {
+        try {
+            ingestionEngine.index(Mockito.any());
+            fail("Expected UnsupportedOperationException to be thrown");
```

Comment:
I think you can use `assertThrows` to actually assert on the type of exception being thrown, which I think would uncover a bug here :)

Reply:
sg